src/protocol/messageSet/decoder.js

```diff
@@ -64,7 +65,20 @@ const EntriesDecoder = (decoder, compressedMessage) => {
 }
 
 const EntryDecoder = decoder => {
+  if (!decoder.canReadInt64()) {
```

Comment:
Won't this mean that we still commit to that offset, even though we didn't get enough bytes to be able to read the last message?

Reply:
I thought the same but I did a check and we commit the offset of the last message

https://github.com/tulios/kafkajs/blob/76bca638368693ffde6caee0ad3d6242ce9cafbe/src/consumer/batch.js#L24

https://github.com/tulios/kafkajs/blob/76bca638368693ffde6caee0ad3d6242ce9cafbe/src/consumer/runner.js#L147

https://github.com/tulios/kafkajs/blob/76bca638368693ffde6caee0ad3d6242ce9cafbe/src/consumer/runner.js#L157

So we are safe on this